metricproviders/kayenta/kayenta.go

```diff
@@ -0,0 +1,202 @@
+
+	JobURL string = "{{inputs.address}}/canary/{{inputs.canaryConfigId}}?application={{inputs.application}}&metricsAccountName={{inputs.metricsAccountName}}&configurationAccountName={{inputs.configurationAccountName}}&storageAccountName={{inputs.storageAccountName}}"
+
+	JobPayloadTemplate string = `
+							{
+								"scopes": {
+										{{inputs.scopes}}
+								},
+                                "thresholds" : {
+                                    "pass": {{inputs.pass}},
+                                    "marginal": {{inputs.marginal}}
+                                }
+                            }`
+
+
+)
+
+type Provider struct {
+	logCtx log.Entry
+	client http.Client
+}
+
+// Type incidates provider is a kayenta provider
+func (p *Provider) Type() string {
+	return ProviderType
+}
+
+// Run queries kayentd for the metric
+func (p *Provider) Run(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric) v1alpha1.Measurement {
+	startTime := metav1.Now()
+	newMeasurement := v1alpha1.Measurement{
+		StartedAt: &startTime,
+	}
+
+	jobURL := strings.Replace(JobURL, "{{inputs.address}}", metric.Provider.Kayenta.Address, 1)
+	jobURL = strings.Replace(jobURL, "{{inputs.canaryConfigId}}", metric.Provider.Kayenta.CanaryConfigId, 1)
+	jobURL = strings.Replace(jobURL, "{{inputs.application}}", metric.Provider.Kayenta.Application, 1)
+	jobURL = strings.Replace(jobURL, "{{inputs.metricsAccountName}}", metric.Provider.Kayenta.MetricsAccountName, 1)
+	jobURL = strings.Replace(jobURL, "{{inputs.configurationAccountName}}", metric.Provider.Kayenta.ConfigurationAccountName, 1)
+	jobURL = strings.Replace(jobURL, "{{inputs.storageAccountName}}", metric.Provider.Kayenta.StorageAccountName, 1)
+
+	jobPayLoad := strings.Replace(JobPayloadTemplate, "{{inputs.pass}}", strconv.Itoa(metric.Provider.Kayenta.Threshold.Pass), 1)
+	jobPayLoad = strings.Replace(jobPayLoad, "{{inputs.marginal}}", strconv.Itoa(metric.Provider.Kayenta.Threshold.Marginal), 1)
+	var scopes string
+	for i, s := range metric.Provider.Kayenta.Scopes {
+		name := s.Name
+		controlScope, err := json.Marshal(s.ControlScope)
+		if err != nil {
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+		experimentScope, err := json.Marshal(s.ExperimentScope)
+		if err != nil {
+			return metricutil.MarkMeasurementError(newMeasurement, err)
+		}
+		scopes = "\"" + name +  "\":" + string(controlScope) + "," + string(experimentScope)
+		if i < (len(metric.Provider.Kayenta.Scopes) - 1) {
+			scopes = scopes + ","
+		}
+
+	}
+
+	jobPayLoad = strings.Replace(jobPayLoad, "{{inputs.scopes}}", scopes, 1)
+
+	jsonValue, err := templateutil.ResolveArgs(jobPayLoad, run.Spec.Arguments)
+	if err != nil {
+		return metricutil.MarkMeasurementError(newMeasurement, err)
+	}
+
+	response, err := p.client.Post(jobURL, "application/json", bytes.NewBuffer([]byte(jsonValue)))
+	if err != nil || response.Body == nil || response.StatusCode > 300 {
```

Comment:
Also, I think the response.body == nil is necessary since I think that will error either at `ioutil.ReadAll(response.Body)` or `err := json.Unmarshal(data, &dat); err != nil`

Reply:
Change to !=200.